radicle-cli/src/git.rs

```diff
@@ -175,24 +175,27 @@ pub fn is_signing_configured(repo: &Path) -> Result<bool, anyhow::Error> {
 }
 
 /// Return the list of radicle remotes for the given repository.
-pub fn remotes(repo: &git2::Repository) -> anyhow::Result<Vec<(String, NodeId)>> {
-    let mut remotes = Vec::new();
-
-    for name in repo.remotes().iter().flatten().flatten() {
-        let remote = repo.find_remote(name)?;
-        for refspec in remote.refspecs() {
-            if refspec.direction() != git2::Direction::Fetch {
-                continue;
-            }
-            if let Some((peer, _)) = refspec.src().and_then(self::parse_remote) {
-                remotes.push((name.to_owned(), peer));
-            }
-        }
-    }
-
+pub fn rad_remotes(repo: &git2::Repository) -> anyhow::Result<Vec<git2::Remote>> {
+    let remotes: Vec<_> = repo
+        .remotes()?
+        .iter()
+        .filter_map(|name| {
+            let remote = repo.find_remote(name?).ok()?;
+            let url = remote.url()?;
+            url.starts_with(radicle::storage::git::transport::local::Url::SCHEME)
+                .then_some(remote)
+        })
+        .collect();
```

Comment:
I'm wondering if we should create our own `Remote` type that looks like this:
```rust
pub struct Remote {
    name: String,
    url: radicle::git::Url,
}
```

That way we can filter out any of the invalid remotes and don't need to do any checks in the `list.rs` module -- all our remotes will have the correct data.

Reply:
Not necessarily true because we can match on the error for parsing the `Url` and if it's `UnsupportedScheme` then it can skip it. Otherwise, it can surface the error for a malformed `Url`.

Reply:
Here I am more inclined with writing a remote wrapper, and move all the necessary code inside an impl, the only reason that I did not  implement this is that there are not another place where we use this, and this look me an optimization, but it’s unclear if it’s yet worthwhile

But looks like a good place for a FIXME and make this refactoring when and if we use this remote struct more often

Reply:
I full agree with this!

Pushed the change here https://github.com/radicle-dev/heartwood/pull/524/commits/f6a496193c3670d69352ce5374147af714c5a42c